### PR TITLE
Support -1 as an alias for dimension size in chunks

### LINKED ISSUE #2689: Short-cut for rechunking an array into a single chunk along an axis?
Currently this requires writing something like `x.rechunk((x.shape[0], ...))` or `x.rechunk((1000000,))`, which is a little clunky.

Possible alternatives:
1. Use `-1` as an alias for "full dimension size", e.g., `x.rechunk((-1,))`.
2. Add a keyword argument, e.g., `rechunk(divisions=(1,))` for rechunking into equally sized divisions instead of chunks.

### PR DESCRIPTION
**Pull request by shoyer**
Fixes #2689

- [x] Tests added / passed
- [x] Passes `flake8 dask`
- [x] Fully documented, including `docs/source/changelog.rst` for all changes
      and one of the `docs/source/*-api.rst` files for new API

### CODE REVIEW

**Comment by mrocklin** [comment]
Fixed the failing test, rebased on master, and force pushed (my apologies for the intrusion into your fork)

**Comment by shoyer** [comment]
> Fixed the failing test, rebased on master, and force pushed (my apologies for the intrusion into your fork)

Thanks!

(This is what happens when I try to bang out a quick PR on the bus without the test suite actually working...)

File: dask/array/core.py [hunk 1]
```diff
@@ -1843,7 +1846,8 @@ def normalize_chunks(chunks, shape=None):
                 "Got chunks=%s, shape=%s" % (chunks, shape))
 
     if shape is not None:
-        chunks = tuple(c if c is not None else s for c, s in zip(chunks, shape))
+        chunks = tuple(c if c not in {None, -1} else s
```
Comment (shoyer) [inline dask/array/core.py]: I also considered using `-1` to indicate "the remaining available size" but the logic there was slightly trickier (and I'm not sure anyone would actually use it).

Comment (mrocklin) [reply dask/array/core.py]: Yeah, I think the current behavior is simple and probably covers the common case.

**Comment by mrocklin** [comment]
Merging soon if there are no further comments

**Pull request merged**